atlante/filestore/file/file.go

```diff
@@ -0,0 +1,115 @@
+package file
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/gdey/errors"
+	"github.com/go-spatial/maptoolkit/atlante/filestore"
+)
+
+const (
+	// TYPE is the name of the provider
+	TYPE = "file"
+
+	// ConfigKeyBasepath is the base directory where the file will be placed.
+	ConfigKeyBasepath = "base_path"
+	// ConfigKeyGroup indicates weather we should group assets in a subdirectory
+	// based on the group name (This is will be the mgdid)
+	ConfigKeyGroup = "group"
+	// ConfigKeyIntermediate is the key used to tell the system to write out the intermediate
+	// files as well.
+	ConfigKeyIntermediate = "intermediate"
```

Comment:
I'm trying to think if the "intermediate" concept should live down at the writer or up a layer. It seems to me this is not something the writer needs to know about but is something that should be implemented higher up so it does not need to be plumbed through to every writer implementation.

Reply:
I think it's important for the writer. It's the writer that determines where files live. So, if I want the intermediate files, I can just add a debug writer that will copy over the intermediate files as well. Instead of having to change all the other settings.